Create Python package

jeangud opened the pull request:
# Description
The current setup seem to require that `satclip` be on a customized `$PYTHONPATH`, for example with:
```shell
export PYTHONPATH=${PYTHONPATH}:/path/to/satclip/satclip/
```

Moreover, it does not list dependencies and versioning (see Issue [Missing requirements.txt file #8](https://github.com/microsoft/satclip/issues/8))

This PR proposes to setup `satclip` as a Python package to install it just like any other Python package, and facilitating its use from any directory, without the need to setup the `PYTHONPATH`.

**Usage:** `pip install .`

**Example use cases:**
- Instantiate the `SatCLIP` PyTorch model: `from satclip.model import SatCLIP`
- Re-use parts of `satclip` in different training scripts `from satclip.location_encoder import LocationEncoder` or again `from satclip.main import SatCLIPLightningModule`

## PR changes
- Added a `pyproject.toml` using [setuptools](https://setuptools.pypa.io/en/latest/index.html#) as a build system, following the latest recommendations from [Python packaging](https://packaging.python.org/en/latest/tutorials/packaging-projects/)
- Lists out main dependencies (albumentations, lightning, rasterio, torchgeo)
- Modified internal imports to be relative to `satclip` package directory

# Testing
- From the repo top directory `pip install .` runs succesfully
- Able to import `satclip` from any folder, without setting `$PYTHONPATH`:
```shell
$ cd ~
$ pwd
/home/user
$ echo $PYTHONPATH

$ python
Python 3.11.0 (main, Jul 29 2024, 13:47:50) [GCC 12.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import satclip
/home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages/kornia/feature/lightglue.py:44: FutureWarning: `torch.cuda.amp.custom_fwd(args...)` is deprecated. Please use `torch.amp.custom_fwd(args..., device_type='cuda')` instead.
  @torch.cuda.amp.custom_fwd(cast_inputs=torch.float32)
>>>
```

<details>
  <summary><b>Example log output</b> (click to expand)</summary>
  Log from pip install .

  ```shell
$ pip install .
Processing /home/user/satclip
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting albumentations (from satclip==0.0.1)
  Using cached albumentations-1.4.13-py3-none-any.whl.metadata (38 kB)
Collecting lightning==2.2.2 (from satclip==0.0.1)
  Using cached lightning-2.2.2-py3-none-any.whl.metadata (53 kB)
Collecting pandas (from satclip==0.0.1)
  Using cached pandas-2.2.2-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (19 kB)
Collecting rasterio>=1.3.10 (from satclip==0.0.1)
  Using cached rasterio-1.3.10-cp311-cp311-manylinux2014_x86_64.whl.metadata (14 kB)
Collecting torchgeo>=0.5 (from satclip==0.0.1)
  Using cached torchgeo-0.5.2-py3-none-any.whl.metadata (20 kB)
Requirement already satisfied: PyYAML<8.0,>=5.4 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from lightning==2.2.2->satclip==0.0.1) (6.0.2)
Requirement already satisfied: fsspec<2025.0,>=2022.5.0 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from fsspec[http]<2025.0,>=2022.5.0->lightning==2.2.2->satclip==0.0.1) (2024.2.0)
Requirement already satisfied: lightning-utilities<2.0,>=0.8.0 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from lightning==2.2.2->satclip==0.0.1) (0.11.6)
Requirement already satisfied: numpy<3.0,>=1.17.2 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from lightning==2.2.2->satclip==0.0.1) (1.26.3)
Requirement already satisfied: packaging<25.0,>=20.0 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from lightning==2.2.2->satclip==0.0.1) (24.1)
Requirement already satisfied: torch<4.0,>=1.13.0 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from lightning==2.2.2->satclip==0.0.1) (2.4.0+cu124)
Requirement already satisfied: torchmetrics<3.0,>=0.7.0 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from lightning==2.2.2->satclip==0.0.1) (1.4.1)
Requirement already satisfied: tqdm<6.0,>=4.57.0 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from lightning==2.2.2->satclip==0.0.1) (4.66.5)
Requirement already satisfied: typing-extensions<6.0,>=4.4.0 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from lightning==2.2.2->satclip==0.0.1) (4.9.0)
Requirement already satisfied: pytorch-lightning in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from lightning==2.2.2->satclip==0.0.1) (2.4.0)
Requirement already satisfied: affine in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from rasterio>=1.3.10->satclip==0.0.1) (2.4.0)
Requirement already satisfied: attrs in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from rasterio>=1.3.10->satclip==0.0.1) (24.2.0)
Requirement already satisfied: certifi in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from rasterio>=1.3.10->satclip==0.0.1) (2024.7.4)
Requirement already satisfied: click>=4.0 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from rasterio>=1.3.10->satclip==0.0.1) (8.1.7)
Requirement already satisfied: cligj>=0.5 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from rasterio>=1.3.10->satclip==0.0.1) (0.7.2)
Requirement already satisfied: snuggs>=1.4.1 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from rasterio>=1.3.10->satclip==0.0.1) (1.4.7)
Requirement already satisfied: click-plugins in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from rasterio>=1.3.10->satclip==0.0.1) (1.1.1)
Requirement already satisfied: setuptools in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from rasterio>=1.3.10->satclip==0.0.1) (65.5.0)
Requirement already satisfied: einops>=0.3 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from torchgeo>=0.5->satclip==0.0.1) (0.8.0)
Requirement already satisfied: fiona>=1.8.19 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from torchgeo>=0.5->satclip==0.0.1) (1.9.6)
Requirement already satisfied: kornia>=0.6.9 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from torchgeo>=0.5->satclip==0.0.1) (0.7.3)
Requirement already satisfied: lightly!=1.4.26,>=1.4.4 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from torchgeo>=0.5->satclip==0.0.1) (1.5.11)
Requirement already satisfied: matplotlib>=3.3.3 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from torchgeo>=0.5->satclip==0.0.1) (3.9.1.post1)
Requirement already satisfied: pillow>=8 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from torchgeo>=0.5->satclip==0.0.1) (10.2.0)
Requirement already satisfied: pyproj>=3 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from torchgeo>=0.5->satclip==0.0.1) (3.6.1)
Requirement already satisfied: rtree>=1 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from torchgeo>=0.5->satclip==0.0.1) (1.3.0)
Requirement already satisfied: segmentation-models-pytorch>=0.2 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from torchgeo>=0.5->satclip==0.0.1) (0.3.3)
Requirement already satisfied: shapely>=1.7.1 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from torchgeo>=0.5->satclip==0.0.1) (2.0.5)
Requirement already satisfied: timm>=0.4.12 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from torchgeo>=0.5->satclip==0.0.1) (0.9.2)
Requirement already satisfied: torchvision>=0.13 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from torchgeo>=0.5->satclip==0.0.1) (0.19.0+cu124)
Requirement already satisfied: python-dateutil>=2.8.2 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from pandas->satclip==0.0.1) (2.9.0.post0)
Requirement already satisfied: pytz>=2020.1 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from pandas->satclip==0.0.1) (2024.1)
Requirement already satisfied: tzdata>=2022.7 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from pandas->satclip==0.0.1) (2024.1)
Requirement already satisfied: scipy>=1.10.0 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from albumentations->satclip==0.0.1) (1.14.0)
Requirement already satisfied: scikit-image>=0.21.0 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from albumentations->satclip==0.0.1) (0.24.0)
Requirement already satisfied: pydantic>=2.7.0 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from albumentations->satclip==0.0.1) (2.8.2)
Requirement already satisfied: albucore>=0.0.13 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from albumentations->satclip==0.0.1) (0.0.13)
Requirement already satisfied: eval-type-backport in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from albumentations->satclip==0.0.1) (0.2.0)
Requirement already satisfied: opencv-python-headless>=4.9.0.80 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from albumentations->satclip==0.0.1) (4.10.0.84)
Requirement already satisfied: tomli>=2.0.1 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from albucore>=0.0.13->albumentations->satclip==0.0.1) (2.0.1)
Requirement already satisfied: six in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from fiona>=1.8.19->torchgeo>=0.5->satclip==0.0.1) (1.16.0)
Requirement already satisfied: aiohttp!=4.0.0a0,!=4.0.0a1 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from fsspec[http]<2025.0,>=2022.5.0->lightning==2.2.2->satclip==0.0.1) (3.10.3)
Requirement already satisfied: kornia-rs>=0.1.0 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from kornia>=0.6.9->torchgeo>=0.5->satclip==0.0.1) (0.1.5)
Requirement already satisfied: hydra-core>=1.0.0 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from lightly!=1.4.26,>=1.4.4->torchgeo>=0.5->satclip==0.0.1) (1.3.2)
Requirement already satisfied: lightly-utils~=0.0.0 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from lightly!=1.4.26,>=1.4.4->torchgeo>=0.5->satclip==0.0.1) (0.0.2)
Requirement already satisfied: requests>=2.23.0 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from lightly!=1.4.26,>=1.4.4->torchgeo>=0.5->satclip==0.0.1) (2.32.3)
Requirement already satisfied: urllib3>=1.25.3 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from lightly!=1.4.26,>=1.4.4->torchgeo>=0.5->satclip==0.0.1) (2.2.2)
Requirement already satisfied: aenum>=3.1.11 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from lightly!=1.4.26,>=1.4.4->torchgeo>=0.5->satclip==0.0.1) (3.1.15)
Requirement already satisfied: bitsandbytes==0.41.0 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from lightning[pytorch-extra]>=2->torchgeo>=0.5->satclip==0.0.1) (0.41.0)
Requirement already satisfied: jsonargparse<5.0,>=4.27.7 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from jsonargparse[signatures]<5.0,>=4.27.7; extra == "pytorch-extra"->lightning[pytorch-extra]>=2->torchgeo>=0.5->satclip==0.0.1) (4.32.0)
Requirement already satisfied: omegaconf<3.0,>=2.0.5 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from lightning[pytorch-extra]>=2->torchgeo>=0.5->satclip==0.0.1) (2.3.0)
Requirement already satisfied: rich<14.0,>=12.3.0 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from lightning[pytorch-extra]>=2->torchgeo>=0.5->satclip==0.0.1) (13.7.1)
Requirement already satisfied: tensorboardX<3.0,>=2.2 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from lightning[pytorch-extra]>=2->torchgeo>=0.5->satclip==0.0.1) (2.6.2.2)
Requirement already satisfied: contourpy>=1.0.1 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from matplotlib>=3.3.3->torchgeo>=0.5->satclip==0.0.1) (1.2.1)
Requirement already satisfied: cycler>=0.10 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from matplotlib>=3.3.3->torchgeo>=0.5->satclip==0.0.1) (0.12.1)
Requirement already satisfied: fonttools>=4.22.0 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from matplotlib>=3.3.3->torchgeo>=0.5->satclip==0.0.1) (4.53.1)
Requirement already satisfied: kiwisolver>=1.3.1 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from matplotlib>=3.3.3->torchgeo>=0.5->satclip==0.0.1) (1.4.5)
Requirement already satisfied: pyparsing>=2.3.1 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from matplotlib>=3.3.3->torchgeo>=0.5->satclip==0.0.1) (3.1.2)
Requirement already satisfied: annotated-types>=0.4.0 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from pydantic>=2.7.0->albumentations->satclip==0.0.1) (0.7.0)
Requirement already satisfied: pydantic-core==2.20.1 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from pydantic>=2.7.0->albumentations->satclip==0.0.1) (2.20.1)
Requirement already satisfied: networkx>=2.8 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from scikit-image>=0.21.0->albumentations->satclip==0.0.1) (3.2.1)
Requirement already satisfied: imageio>=2.33 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from scikit-image>=0.21.0->albumentations->satclip==0.0.1) (2.34.2)
Requirement already satisfied: tifffile>=2022.8.12 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from scikit-image>=0.21.0->albumentations->satclip==0.0.1) (2024.8.10)
Requirement already satisfied: lazy-loader>=0.4 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from scikit-image>=0.21.0->albumentations->satclip==0.0.1) (0.4)
Requirement already satisfied: pretrainedmodels==0.7.4 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from segmentation-models-pytorch>=0.2->torchgeo>=0.5->satclip==0.0.1) (0.7.4)
Requirement already satisfied: efficientnet-pytorch==0.7.1 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from segmentation-models-pytorch>=0.2->torchgeo>=0.5->satclip==0.0.1) (0.7.1)
Requirement already satisfied: huggingface-hub in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from timm>=0.4.12->torchgeo>=0.5->satclip==0.0.1) (0.24.5)
Requirement already satisfied: safetensors in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from timm>=0.4.12->torchgeo>=0.5->satclip==0.0.1) (0.4.4)
Requirement already satisfied: munch in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from pretrainedmodels==0.7.4->segmentation-models-pytorch>=0.2->torchgeo>=0.5->satclip==0.0.1) (4.0.0)
Requirement already satisfied: filelock in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from torch<4.0,>=1.13.0->lightning==2.2.2->satclip==0.0.1) (3.13.1)
Requirement already satisfied: sympy in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from torch<4.0,>=1.13.0->lightning==2.2.2->satclip==0.0.1) (1.12)
Requirement already satisfied: jinja2 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from torch<4.0,>=1.13.0->lightning==2.2.2->satclip==0.0.1) (3.1.3)
Requirement already satisfied: nvidia-cuda-nvrtc-cu12==12.4.99 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from torch<4.0,>=1.13.0->lightning==2.2.2->satclip==0.0.1) (12.4.99)
Requirement already satisfied: nvidia-cuda-runtime-cu12==12.4.99 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from torch<4.0,>=1.13.0->lightning==2.2.2->satclip==0.0.1) (12.4.99)
Requirement already satisfied: nvidia-cuda-cupti-cu12==12.4.99 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from torch<4.0,>=1.13.0->lightning==2.2.2->satclip==0.0.1) (12.4.99)
Requirement already satisfied: nvidia-cudnn-cu12==9.1.0.70 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from torch<4.0,>=1.13.0->lightning==2.2.2->satclip==0.0.1) (9.1.0.70)
Requirement already satisfied: nvidia-cublas-cu12==12.4.2.65 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from torch<4.0,>=1.13.0->lightning==2.2.2->satclip==0.0.1) (12.4.2.65)
Requirement already satisfied: nvidia-cufft-cu12==11.2.0.44 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from torch<4.0,>=1.13.0->lightning==2.2.2->satclip==0.0.1) (11.2.0.44)
Requirement already satisfied: nvidia-curand-cu12==10.3.5.119 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from torch<4.0,>=1.13.0->lightning==2.2.2->satclip==0.0.1) (10.3.5.119)
Requirement already satisfied: nvidia-cusolver-cu12==11.6.0.99 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from torch<4.0,>=1.13.0->lightning==2.2.2->satclip==0.0.1) (11.6.0.99)
Requirement already satisfied: nvidia-cusparse-cu12==12.3.0.142 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from torch<4.0,>=1.13.0->lightning==2.2.2->satclip==0.0.1) (12.3.0.142)
Requirement already satisfied: nvidia-nccl-cu12==2.20.5 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from torch<4.0,>=1.13.0->lightning==2.2.2->satclip==0.0.1) (2.20.5)
Requirement already satisfied: nvidia-nvtx-cu12==12.4.99 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from torch<4.0,>=1.13.0->lightning==2.2.2->satclip==0.0.1) (12.4.99)
Requirement already satisfied: nvidia-nvjitlink-cu12==12.4.99 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from torch<4.0,>=1.13.0->lightning==2.2.2->satclip==0.0.1) (12.4.99)
Requirement already satisfied: triton==3.0.0 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from torch<4.0,>=1.13.0->lightning==2.2.2->satclip==0.0.1) (3.0.0)
Requirement already satisfied: aiohappyeyeballs>=2.3.0 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from aiohttp!=4.0.0a0,!=4.0.0a1->fsspec[http]<2025.0,>=2022.5.0->lightning==2.2.2->satclip==0.0.1) (2.3.5)
Requirement already satisfied: aiosignal>=1.1.2 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from aiohttp!=4.0.0a0,!=4.0.0a1->fsspec[http]<2025.0,>=2022.5.0->lightning==2.2.2->satclip==0.0.1) (1.3.1)
Requirement already satisfied: frozenlist>=1.1.1 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from aiohttp!=4.0.0a0,!=4.0.0a1->fsspec[http]<2025.0,>=2022.5.0->lightning==2.2.2->satclip==0.0.1) (1.4.1)
Requirement already satisfied: multidict<7.0,>=4.5 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from aiohttp!=4.0.0a0,!=4.0.0a1->fsspec[http]<2025.0,>=2022.5.0->lightning==2.2.2->satclip==0.0.1) (6.0.5)
Requirement already satisfied: yarl<2.0,>=1.0 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from aiohttp!=4.0.0a0,!=4.0.0a1->fsspec[http]<2025.0,>=2022.5.0->lightning==2.2.2->satclip==0.0.1) (1.9.4)
Requirement already satisfied: antlr4-python3-runtime==4.9.* in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from hydra-core>=1.0.0->lightly!=1.4.26,>=1.4.4->torchgeo>=0.5->satclip==0.0.1) (4.9.3)
Requirement already satisfied: docstring-parser>=0.15 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from jsonargparse[signatures]<5.0,>=4.27.7; extra == "pytorch-extra"->lightning[pytorch-extra]>=2->torchgeo>=0.5->satclip==0.0.1) (0.16)
Requirement already satisfied: typeshed-client>=2.1.0 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from jsonargparse[signatures]<5.0,>=4.27.7; extra == "pytorch-extra"->lightning[pytorch-extra]>=2->torchgeo>=0.5->satclip==0.0.1) (2.7.0)
Requirement already satisfied: charset-normalizer<4,>=2 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from requests>=2.23.0->lightly!=1.4.26,>=1.4.4->torchgeo>=0.5->satclip==0.0.1) (3.3.2)
Requirement already satisfied: idna<4,>=2.5 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from requests>=2.23.0->lightly!=1.4.26,>=1.4.4->torchgeo>=0.5->satclip==0.0.1) (3.7)
Requirement already satisfied: markdown-it-py>=2.2.0 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from rich<14.0,>=12.3.0->lightning[pytorch-extra]>=2->torchgeo>=0.5->satclip==0.0.1) (3.0.0)
Requirement already satisfied: pygments<3.0.0,>=2.13.0 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from rich<14.0,>=12.3.0->lightning[pytorch-extra]>=2->torchgeo>=0.5->satclip==0.0.1) (2.18.0)
Requirement already satisfied: protobuf>=3.20 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from tensorboardX<3.0,>=2.2->lightning[pytorch-extra]>=2->torchgeo>=0.5->satclip==0.0.1) (5.27.3)
Requirement already satisfied: MarkupSafe>=2.0 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from jinja2->torch<4.0,>=1.13.0->lightning==2.2.2->satclip==0.0.1) (2.1.5)
Requirement already satisfied: mpmath>=0.19 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from sympy->torch<4.0,>=1.13.0->lightning==2.2.2->satclip==0.0.1) (1.3.0)
Requirement already satisfied: mdurl~=0.1 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from markdown-it-py>=2.2.0->rich<14.0,>=12.3.0->lightning[pytorch-extra]>=2->torchgeo>=0.5->satclip==0.0.1) (0.1.2)
Requirement already satisfied: importlib-resources>=1.4.0 in /home/user/.pyenv/versions/3.11.0/lib/python3.11/site-packages (from typeshed-client>=2.1.0->jsonargparse[signatures]<5.0,>=4.27.7; extra == "pytorch-extra"->lightning[pytorch-extra]>=2->torchgeo>=0.5->satclip==0.0.1) (6.4.0)
Using cached lightning-2.2.2-py3-none-any.whl (2.0 MB)
Using cached rasterio-1.3.10-cp311-cp311-manylinux2014_x86_64.whl (21.5 MB)
Using cached torchgeo-0.5.2-py3-none-any.whl (381 kB)
Using cached pandas-2.2.2-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (13.0 MB)
Using cached albumentations-1.4.13-py3-none-any.whl (171 kB)
Building wheels for collected packages: satclip
  Building wheel for satclip (pyproject.toml) ... done
  Created wheel for satclip: filename=satclip-0.0.1-py3-none-any.whl size=4017354 sha256=4e4af561efcaea8c268b353480d7a87f575904e9055dc0b653f15544292537c1
  Stored in directory: /tmp/pip-ephem-wheel-cache-fal20ggn/wheels/3e/73/bd/53cd44af9d38c274b7928db633e7a5b0a095d97c6ad8fe0686
Successfully built satclip
Installing collected packages: rasterio, pandas, albumentations, lightning, torchgeo, satclip
Successfully installed albumentations-1.4.13 lightning-2.2.2 pandas-2.2.2 rasterio-1.3.10 satclip-0.0.1 torchgeo-0.5.2
  ```
</details>

jeangud committed:
Added packaging files

jeangud commented on pyproject.toml:
```diff
@@ -0,0 +1,39 @@
+[build-system]
```
This file is the main change in this PR

konstantinklemmer commented:
Thank you so much for this @jeangud! Just a quick note: I am taking some time off currently but will get to this as soon as I am back to working. Thanks for your contributions to this project.

konstantinklemmer committed:
Merge branch 'main' into packaging

Bring packaging branch up to date with latest main (README dataset/model
updates and notebook URL fixes).

konstantinklemmer committed:
Polish packaging: docs, deps, configs, entry point, CI

Build on the packaging work with the follow-ups needed to make the package
usable end-to-end:

- docs/notebooks: migrate README quickstart and notebooks (A01/A02/B01/B02)
  off 'git clone + sys.path' to 'pip install' + 'from satclip... import ...';
  add an Installation section and run training via 'python -m satclip.main'
- pyproject: list direct deps explicitly (torch, numpy, sympy, timm, einops,
  matplotlib, torchvision, huggingface_hub), relax 'lightning' pin to >=2.2,<3,
  SPDX license, add 'examples'/'dev' extras and a 'satclip-train' entry point,
  ship configs/*.yaml as package data
- main.py: resolve the default config relative to the package and guard the
  CUDA device check so 'python -m satclip.main' works from any directory / on CPU
- __init__.py: expose a clean public API (SatCLIP, LocationEncoder,
  SatCLIPLoss, get_satclip) instead of wildcard re-exports
- add a CI workflow and import/forward-pass smoke tests
- ignore dist/ and .pytest_cache/

Verified: wheel builds, bundles configs/default.yaml, registers the
satclip-train entry point, and emits License-Expression: MIT.
